src/HealthChecks.CosmosDb/TableServiceHealthCheck.cs

```diff
@@ -8,72 +6,51 @@ namespace HealthChecks.CosmosDb
+                  new TableServiceHealthCheckOptions { TableName = tableName })
+        { }
+
+        public TableServiceHealthCheck(TableServiceClient tableServiceClient, TableServiceHealthCheckOptions options)
         {
-            _connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
-            _tableName = tableName ?? throw new ArgumentNullException(nameof(tableName));
-        }
-
-        public TableServiceHealthCheck(Uri endpoint, TableSharedKeyCredential credentials, string tableName)
-        {
-            _endpoint = endpoint ?? throw new ArgumentNullException(nameof(endpoint));
-            _credentials = credentials ?? throw new ArgumentNullException(nameof(credentials));
-            _tableName = tableName ?? throw new ArgumentNullException(nameof(tableName));
-        }
-
-        public TableServiceHealthCheck(Uri endpoint, TokenCredential tokenCredential, string tableName)
-        {
-            _endpoint = endpoint ?? throw new ArgumentNullException(nameof(endpoint));
-            _tokenCredential = tokenCredential ?? throw new ArgumentNullException(nameof(tokenCredential));
-            _tableName = tableName ?? throw new ArgumentNullException(nameof(tableName));
+            _tableServiceClient = tableServiceClient ?? throw new ArgumentNullException(nameof(tableServiceClient));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
         }
 
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
             try
             {
-                var tableServiceKey = _connectionString ?? _endpoint!.ToString();
-                if (!_connections.TryGetValue(tableServiceKey, out var tableServiceClient))
-                {
-                    tableServiceClient = CreateTableServiceClient();
+                await _tableServiceClient.GetPropertiesAsync(cancellationToken);
 
-                    if (!_connections.TryAdd(tableServiceKey, tableServiceClient))
-                    {
-                        tableServiceClient = _connections[tableServiceKey];
-                    }
+                if (!string.IsNullOrEmpty(_options.TableName))
+                {
+                    var tableClient = _tableServiceClient.GetTableClient(_options.TableName);
+                    await tableClient.GetAccessPoliciesAsync(cancellationToken);
                 }
-                var tableClient = tableServiceClient.GetTableClient(_tableName);
-                _ = await tableClient.GetAccessPoliciesAsync();
 
                 return HealthCheckResult.Healthy();
             }
-            catch (RequestFailedException ex) when (ex.Status == 404)
-            {
-                return new HealthCheckResult(context.Registration.FailureStatus, description: $"Table with name {_tableName} does not exist.");
-            }

```

Comment:
Why?

Reply:
Similar to the other Azure Storage services, a 404 will also result in an exception whose message will be like `"The table specified does not exist.<more request info>"` so I think special-casing this scenario is superfluous